ghw/libghw.c

```diff
@@ -1078,13 +1171,14 @@ ghw_read_hie (struct ghw_handler *h)
   /* Number of declared signals (which may be composite).  */
   nbr_sigs = ghw_get_i32 (h, &hdr[8]);
   /* Number of basic signals.  */
-  h->nbr_sigs = ghw_get_i32 (h, &hdr[12]);
+  h->nbr_sigs = (uint32_t) ghw_get_i32 (h, &hdr[12]);
+  if (h->nbr_sigs > UINT32_MAX - 10) ghw_error_exit();
```

Comment:
Check for non-negative value instead (maybe we should create `ghw_get_i32`.

Reply:
Done, I've added a function `ghw_get_i32_positive`.